### PR TITLE
Removed unused template multimedia_uploader.html

### PR DESCRIPTION
**Pull request by orangejenny**
## Technical Summary
Minor cleanup as I'm working on YUI deprecation.

### Safety story
Dead code removal, code review is a sufficient safety check.

### Automated test coverage

no

### QA Plan

Not requesting QA. I smoke tested uploading a menu icon.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/hqmedia/templates/hqmedia/partials/hqm_upload_form.html [hunk 1]
```diff
@@ -1,29 +0,0 @@
-{% load i18n %}
-<fieldset class="hqm-sharing hide">
```
Comment (orangejenny) [inline corehq/apps/hqmedia/templates/hqmedia/partials/hqm_upload_form.html:2]: This is all hidden by default due to the `hide` class. The way to show it is via the "share media" checkbox (see [here](https://github.com/dimagi/MediaUploader/blob/1038b486ecfca4251bb5c85bc427a92df35cf8d7/hqmedia.upload_controller.js#L211-L214)), but HQ doesn't have that checkbox. It did a long time ago (see [here](https://github.com/dimagi/commcare-hq/commit/eb3f82f2df7be1e07660b10f8b5b99f8147dbc0b#diff-24d4c57eafa007ed37b5fad665493bda8ebdeec7a33470d87efed3928560ff1cR9)) but it was removed at some point.

File: corehq/apps/hqmedia/templates/hqmedia/partials/multimedia_uploader.html [hunk 2]
```diff
@@ -27,9 +27,6 @@ <h3 class="modal-title">{% trans "Upload" %} {{ uploader.media_type|title }}</h3
           </label>
           <div class="hqm-existing-controls col-sm-9"></div>
         </div>
-        <div class="hqm-upload-form">
-          {% include 'hqmedia/partials/hqm_upload_form.html' %}
-        </div>
```
Comment (mkangia) [inline corehq/apps/hqmedia/templates/hqmedia/partials/multimedia_uploader.html:32]: Is multimedia_uploader entirely not in use anymore? but we are removing just a part of it?

Comment (orangejenny) [reply corehq/apps/hqmedia/templates/hqmedia/partials/multimedia_uploader.html:32]: This just removes a part of it, although I'm working on removing it altogether.

**Review by millerdev — APPROVED** [review]

File: corehq/apps/hqmedia/templates/hqmedia/partials/multimedia_uploader.html (hunk 2 above)
Comment (mkangia) [reply corehq/apps/hqmedia/templates/hqmedia/partials/multimedia_uploader.html:32]: okay thank you. It was not clear to me how we determined this part was safe to remove. I thought we don't have uploaders anymore, but I could not determine that from the views.

Comment (orangejenny) [reply corehq/apps/hqmedia/templates/hqmedia/partials/multimedia_uploader.html:32]: The uploader code is rather...layered. I'm still getting my head around it. The functionality removed in this particular PR is something I also removed in Vellum, [here](https://github.com/dimagi/Vellum/pull/1122/commits/48eb028dfdd363b8475bc130551e770f68a027db), so I just verified that it was safe to make the same change in HQ (and I think it's safe, just based on [this](https://github.com/dimagi/commcare-hq/pull/35181#discussion_r1790557417)).

Comment (mkangia) [reply corehq/apps/hqmedia/templates/hqmedia/partials/multimedia_uploader.html:32]: Cool, that is helpful :+1:

**Pull request merged**